packages/uhhyou-plugins/PKGBUILD

```diff
@@ -0,0 +1,79 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Florian Hülsmann <fh@cbix.de>
+
+_name=VSTPlugins-UhhyouPlugins
+pkgname=uhhyou-plugins
+pkgver=0.49.0
+pkgrel=1
+pkgdesc='A collection of synthesizer and effect VST3 plugins'
+arch=(x86_64)
+url='https://ryukau.github.io/VSTPlugins/'
+license=(GPL3)
+groups=(pro-audio vst3-plugins)
+depends=(xcb-util-cursor)
+makedepends=(cairo cmake freetype2 gtkmm3 libx11 libxkbcommon-x11 pango vst3sdk xcb-util xcb-util-keysyms)
+optdepends=('vst3-host: for running VST3 plugins')
+_pocketfft_ref=53e9dd4
+_vcl_ref=46a5841
+source=(
+  "$pkgname-$pkgver.tar.gz::https://github.com/ryukau/VSTPlugins/archive/refs/tags/UhhyouPlugins$pkgver.tar.gz"
+  "pocketfft-$_pocketfft_ref.tar.gz::https://gitlab.mpcdf.mpg.de/mtr/pocketfft/-/archive/$_pocketfft_ref/pocketfft-$_pocketfft_ref.tar.gz"
+  "vcl-$_vcl_ref.tar.gz::https://github.com/vectorclass/version2/archive/$_vcl_ref.tar.gz"
+)
+sha256sums=('0ab6d959baf932633fcaad064731da4721c9247b95ae2d2f433ba5cd612bc955'
+            '0f2b09e14b1497c9a1283c5b64a6682fd43daa6ba7ca17eb3dbcc9b0ec46204f'
+            'a7f5b8c5a311ecc6d969012a6e15115a41f8f135862e03291d8b5a12f9ee9463')
+
+prepare() {
+  cd $_name$pkgver
+  # link submodules
+  rm -r lib/pocketfft && ln -s "$srcdir"/pocketfft-$_pocketfft_ref lib/pocketfft
+  rm -r lib/vcl && ln -s "$srcdir"/version2-$_vcl_ref* lib/vcl
+  # copy and patch vst3sdk
+  cp -r /usr/share/vst3sdk ../
+  cp ci/linux_patch/cairocontext.cpp ../vst3sdk/vstgui4/vstgui/lib/platform/linux/cairocontext.cpp
+}
+
+build() {
+  cmake -B build -S vst3sdk \
+    -DCMAKE_BUILD_TYPE=Release \
```

Comment:
Are you sure "Release" is the best option?

The Arch packaging guidelines recommend "None":

https://wiki.archlinux.org/title/CMake_package_guidelines#CMake_undesired_behaviors

... but I know for performance critical software like this, this might not be the best option.

So, just asking whether this an informed decision?

Reply:
Yes, the build types are adding `-O3 -fPIC`
- O3 may be necessary for auto-vectorization, not 100% sure here
- PIC should probably be added independent of the build type but for some reason they're doing it here

Reply:
I'll try building with None and test if it makes any difference

Reply:
Apparently, vst3sdk fails on build type `None`:
```
[  1%] Building CXX object base/CMakeFiles/base.dir/source/fbuffer.cpp.o
In file included from /home/florian/build/archlinux-proaudio/packages/uhhyou-plugins/src/vst3sdk/base/source/fobject.h:46,
                 from /home/florian/build/archlinux-proaudio/packages/uhhyou-plugins/src/vst3sdk/base/source/fstring.h:45,
                 from /home/florian/build/archlinux-proaudio/packages/uhhyou-plugins/src/vst3sdk/base/source/fbuffer.cpp:39:
/home/florian/build/archlinux-proaudio/packages/uhhyou-plugins/src/vst3sdk/base/source/fdebug.h:72:18: error: #error DEVELOPMENT, RELEASE, _DEBUG, or NDEBUG must be defined!
   72 |                 #error DEVELOPMENT, RELEASE, _DEBUG, or NDEBUG  must be defined!
      |                  ^~~~~
make[2]: *** [base/CMakeFiles/base.dir/build.make:90: base/CMakeFiles/base.dir/source/fbuffer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1164: base/CMakeFiles/base.dir/all] Error 2
```

Reply:
Ok, "Release" it is then.